src/codeu/chat/client/core/Controller.java

```diff
@@ -34,11 +35,17 @@
   private final static Logger.Log LOG = Logger.newLog(Controller.class);
 
   private final ConnectionSource source;
+  private User user;
 
   public Controller(ConnectionSource source) {
     this.source = source;
   }
 
+  public Controller(Controller controller, User user) {
+    this(controller.source);
```

Comment:
Actually, this is part of a major hack to instantiate a new Controller and View but with an actual user attached to them so that the user id can be sent over the wire.